cache/compression/compression.go

```diff
@@ -0,0 +1,105 @@
+package compression
+
+import (
+	"errors"
+	"io"
+	"io/fs"
+	"os"
+
+	"github.com/bitrise-io/go-utils/v2/command"
+	"github.com/bitrise-io/go-utils/v2/env"
+	"github.com/bitrise-io/go-utils/v2/log"
+)
+
+// Compress creates a compressed archive from the provided files and folders using absolute paths.
+func Compress(archivePath string, includePaths []string, logger log.Logger, envRepo env.Repository) error {
+	cmdFactory := command.NewFactory(envRepo)
+
+	tarArgs := []string{
+		"--use-compress-program",
+		"zstd --threads=0", // Use CPU count threads
+		"-P",               // Same as --absolute-paths in BSD tar, --absolute-names in GNU tar
+		"-cf",
+		archivePath,
+		"--directory",
+		envRepo.Get("BITRISE_SOURCE_DIR"),
+	}
+	tarArgs = append(tarArgs, includePaths...)
+
+	cmd := cmdFactory.Create("tar", tarArgs, nil)
+
+	logger.Debugf("$ %s", cmd.PrintableCommandArgs())
+
+	out, err := cmd.RunAndReturnTrimmedCombinedOutput()
+	if err != nil {
+		logger.Errorf("Compression command failed: %s", out)
+		return err
+	}
+
+	return nil
+}
+
+// Decompress takes an archive path and extracts files. This assumes an archive created with absolute file paths.
+func Decompress(archivePath string, logger log.Logger, envRepo env.Repository, additionalArgs ...string) error {
+	commandFactory := command.NewFactory(envRepo)
+
+	decompressTarArgs := []string{
+		"--use-compress-program",
+		"zstd -d",
+		"-xf",
+		archivePath,
+		"-P", // Same as --absolute-paths in BSD tar, --absolute-names in GNU tar
+	}
+
+	if len(additionalArgs) > 0 {
+		decompressTarArgs = append(decompressTarArgs, additionalArgs...)
+	}
+
+	cmd := commandFactory.Create("tar", decompressTarArgs, nil)
+	logger.Debugf("$ %s", cmd.PrintableCommandArgs())
+
+	output, err := cmd.RunAndReturnTrimmedCombinedOutput()
+	if err != nil {
+		logger.Errorf("Failed to decompress cache archive: %s", output)
+		return err
+	}
+
+	return nil
+}
+
+func areAllPathsEmpty(includePaths []string) bool {
```

Comment:
This private function is only used in the test as part of the assertion. I would move this into the test file just to visually indicate that they belong together. (Unless there will be a use case coming in one of the following PRs.)

Reply:
Good catch! This is unused for now, the code that calls this is not moved over yet: https://github.com/bitrise-steplib/bitrise-step-save-cache/blob/7f51168e673fe0794d84977dfdec1664cc49f8e1/step/step.go#L205

I decided to make it private for now, but let me know if you'd like to make it public right now.

Reply:
It is perfectly fine if it is private for now.

Reply:
@godrei the answer for your question in the other thread